glue_jupyter/state_traitlets_helpers.py

```diff
@@ -45,12 +45,14 @@ def update_state_from_dict(state, changes):
             if isinstance(getattr(state, name), CallbackList):
                 callback_list = getattr(state, name)
                 for i in range(len(callback_list)):
-                    if isinstance(callback_list[i], State):
-                        update_state_from_dict(callback_list[i], changes[name][i])
-                    else:
-                        if (changes[name][i] != MAGIC_IGNORE and
-                                callback_list[i] != changes[name][i]):
-                            callback_list[i] = changes[name][i]
+                    if (isinstance(changes[name], dict) and i in changes[name]
```

Comment:
`changes[name]` can only be an index-dict here (as in the test), right? Should perhaps be noted somewhere for clarity.

Reply:
Done